VideoLocker/src/main/java/org/edx/mobile/view/CourseOutlineFragment.java

```diff
@@ -183,4 +193,58 @@ public void reloadList(){
             adapter.reloadData();
         }
     }
+
+    private void updateRowSelection(String lastAccessedId){
+        if (!TextUtils.isEmpty(lastAccessedId)) {
+            final int selectedItemPosition = adapter.getPositionByItemId(lastAccessedId);
+            if (selectedItemPosition != -1) {
+                listView.setItemChecked(selectedItemPosition, true);
+                listView.postDelayed(new Runnable() {
+                    @Override
+                    public void run() {
+                        listView.smoothScrollToPosition(selectedItemPosition);
+                    }
+                }, AUTOSCROLL_DELAY_MS);
+            }
+        }
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        switch (requestCode) {
+            // If user has navigated to a different unit, then we need to rearrange
+            // the activity stack to point to it.
+            case REQUEST_SHOW_COURSE_UNIT_DETAIL: {
+                switch (resultCode) {
+                    case Activity.RESULT_OK: {
+                        CourseComponent outlineComp = courseManager.getComponentById(
+                                courseData.getCourse().getId(), courseComponentId);
+                        String leafCompId = (String) data.getSerializableExtra(Router.EXTRA_COURSE_COMPONENT_ID);
+                        CourseComponent leafComp = courseManager.getComponentById(
+                                courseData.getCourse().getId(), leafCompId);
+                        BlockPath outlinePath = outlineComp.getPath();
+                        BlockPath leafPath = leafComp.getPath();
+                        int outlinePathSize = outlinePath.getPath().size();
+                        if (!outlineComp.equals(leafPath.get(outlinePathSize - 1))) {
+                            getActivity().setResult(Activity.RESULT_OK, data);
+                            getActivity().finish();
```

Comment:
The `Activity` should be initialized to a variable outside the scope of the if statement, to avoid calling it multiple times unnecessarily.